### PR TITLE
Updated durable functions templates in C# to use nameof 

### PR DESCRIPTION
**Pull request by lbugnion**
This uses nameof wherever possible instead of magic strings. This makes navigating to the symbols (F12 in Visual Studio) much easier and more convenient. It also avoids errors when renaming a method.

### CODE REVIEW

**Comment by anthonychu** [comment]
Thanks @lbugnion.

cc @cgillum @ConnorMcMahon

**Review by liliankasem — APPROVED** [review]

**Comment by davidmrdavid** [comment]
Thanks for your contribution. This seems right to me but, before merging, I think it'd be good to check-in with @AnatoliB and @cgillum as to whether or not we want to incorporate this pattern into the templates. If possible, I'd like to wait until they can give this an approval themselves :)

**Comment by liliankasem** [comment]
> Thanks for your contribution. This seems right to me but, before merging, I think it'd be good to check-in with @AnatoliB and @cgillum as to whether or not we want to incorporate this pattern into the templates. If possible, I'd like to wait until they can give this an approval themselves :)

@davidmrdavid @AnatoliB  @cgillum - any thoughts on this before we merge?

**Comment by davidmrdavid** [comment]
My advice still stands: I like the change, but I want to get at least one more DF person to approve since this affects our quickstart experience. I'll wait for either of the tagged folks above to chime in :-)

**Review by AnatoliB — APPROVED** [review]
I like this change.

**Review by cgillum — APPROVED** [review]
LGTM! This is definitely a best practice, so having it in our templates feels like the right thing to do.

**Comment by davidmrdavid** [comment]
@liliankasem go ahead and merge then!

**Pull request merged**
